### DISCUSSION TITLE
How to get parent for Node?

### DISCUSSION THREAD

**Discussion by evidolob** [eemeli/yaml#297] [Q&A] [answered]
I'm currently implementing code completion for yaml-language-server with `yaml` parser.
I can found AST node on which code completion was executed, but in some cases I need to get parent Node for it.
Is there any possibility to get parent node?

**Comment by eemeli** [comment, marked as answer]
At least for now that's not directly available from a `Node`, as they do not maintain parent links. If you have a reference for the node's `Document`, then the easiest way to find the parent is almost certainly to `visit()` the doc, terminating the iteration when you find it.

**Reply by evidolob** [reply]
@eemeli Thx.
